saleor/graphql/attribute/mutations/attribute_bulk_create.py

```diff
@@ -0,0 +1,548 @@
+        required=False,
+    )
+    file_url = graphene.String(
+        required=False,
+        description="URL of the file attribute. Every time, a new value is created.",
+    )
+    content_type = graphene.String(required=False, description="File content type.")
+
+    class Meta:
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+
+class AttributeBulkCreateInput(BaseInputObjectType):
+    external_reference = graphene.String(
+        description="External ID of this attribute.", required=False
+    )
+    input_type = AttributeInputTypeEnum(description=AttributeDescriptions.INPUT_TYPE)
+    entity_type = AttributeEntityTypeEnum(description=AttributeDescriptions.ENTITY_TYPE)
+    name = graphene.String(required=True, description=AttributeDescriptions.NAME)
+    slug = graphene.String(required=False, description=AttributeDescriptions.SLUG)
+    type = AttributeTypeEnum(description=AttributeDescriptions.TYPE, required=True)
+    unit = MeasurementUnitsEnum(description=AttributeDescriptions.UNIT, required=False)
+    values = NonNullList(
+        AttributeBulkCreateValueInput, description=AttributeDescriptions.VALUES
+    )
+    value_required = graphene.Boolean(description=AttributeDescriptions.VALUE_REQUIRED)
+    is_variant_only = graphene.Boolean(
+        required=False, description=AttributeDescriptions.IS_VARIANT_ONLY
+    )
+    visible_in_storefront = graphene.Boolean(
+        description=AttributeDescriptions.VISIBLE_IN_STOREFRONT
+    )
+
+    filterable_in_dashboard = graphene.Boolean(
+        description=AttributeDescriptions.FILTERABLE_IN_DASHBOARD
+    )
+
+    class Meta:
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+
+class AttributeBulkCreate(BaseMutation):
+    count = graphene.Int(
+        required=True,
+        description="Returns how many objects were created.",
+    )
+    results = NonNullList(
+        AttributeBulkResult,
+        required=True,
+        default_value=[],
+        description="List of the created attributes.",
+    )
+
+    class Arguments:
+        attributes = NonNullList(
+            AttributeBulkCreateInput,
+            required=True,
+            description="Input list of products to create.",
+        )
+        error_policy = ErrorPolicyEnum(
+            required=False,
+            description="Policies of error handling. DEFAULT: "
+            + ErrorPolicyEnum.REJECT_EVERYTHING.name,
+        )
+
+    class Meta:
+        description = "Creates attributes." + ADDED_IN_315 + PREVIEW_FEATURE
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+        error_type_class = AttributeBulkCreateError
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.ATTRIBUTE_CREATED,
+                description="An attribute was created.",
+            ),
+        ]
+
+    @classmethod
+    def clean_attributes(cls, info, attributes_data, index_error_map):
+        cleaned_inputs_map: dict = {}
+
+        attrs_existing_external_refs = set(
```

Comment:
Are you aware that you're fetching here data from **ALL** Attribute, AttributeValue objects?This won't work in case of having a huge dataset of the attributes

Reply:
Yep, but do you have any other ideas how to validate if external ref is unique before save? @korycins

Reply:
Maybe we could do it without loading the entire dataset into memory and check it using a database query with filter and "exists" query?

Reply:
but this will generate query for every attribute and attribute value from input, so if there will be 200 objects with external refs we will have 200 queries just for checking external ref

Reply:
You can make one query using `IN`, or I'm missing something?

Reply:
@kadewu but `exists()` returns only `True/False` so I won't be able to check which input has invalid external_ref